### PR TITLE
Add device.architecture to logs

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What does this PR do?

This adds a `device.architecture` property to logs. Device is kept as an object for potential future expansion

### Motivation

The device architecture is needed for Flutter symbolication of logs.

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Review by mariusc83 — APPROVED** [review]

File: instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt [hunk 1]
```diff
@@ -92,7 +92,7 @@ internal object RuntimeConfig {
 
     val stringAttributes = mapOf(
         "version_name" to BuildConfig.VERSION_NAME,
-        "device" to Build.DEVICE,
+        "build_device" to Build.DEVICE,
```
Comment (0xnm) [inline instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt:95]: should we revert this change now?

**Review by 0xnm — APPROVED** [review]

**Review by mariusc83 — APPROVED** [review]

File: instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt (hunk 1 above)
Comment (fuzzybinary) [reply instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt:95]: Yes, good catch. I actually thought I did revert it.

**Review by xgouchet — APPROVED** [review]
